### PR TITLE
Add VALUES as top level relation

### PR DESCRIPTION
**Pull request by mfussenegger**
## Summary of the changes / Why this improves CrateDB

Besides improving SQL Standard compatiblity, this will allow us to
de-duplicate the INSERT FROM VALUES / INSERT FROM query logic by
treating VALUES as a regular relation.

Internally it is currently being re-written to ``UNNEST`` as it lets us
avoid having to add a new operator, execution phase, execution plan and
execution task. The downside is that we've a row-orientation ->
column-orientation -> row-orientation roundrip.

We can later on optimize that if it turns out to be necessary.

## Checklist

 - [x] User relevant changes are recorded in ``CHANGES.txt``
 - [x] Touched code is covered by tests
 - [x] Documentation has been updated if necessary
 - [x] [CLA](https://crate.io/community/contribute/cla/) is signed
 - [x] This does not contain breaking changes, or if it does:
    - It is released within a major release
    - It is recorded in ``CHANGES.txt``
    - It was marked as deprecated in an earlier release if possible
    - You've thought about the consequences and other components are adapted
      (E.g. AdminUI)

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -32,6 +32,7 @@ __pycache__/
 *.pyc
 *.pyo
 blackbox/get-pip.py
+blackbox/.mypy_cache/
```
Comment (andreidan) [inline .gitignore]: uber minor but should this be in a separate commit?

File: blackbox/docs/sql/statements/values.rst [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+   |    2 | two   |
+   |    3 | three |
+   +------+-------+
+   VALUES 3 rows in set (... sec)
+
+
+It is commonly used in :ref:`ref-insert` to provide values to insert into a
+table.
+
+The expressions across rows that make up the same columns must have the same
```
Comment (andreidan) [inline blackbox/docs/sql/statements/values.rst]: rows and columns need to be swapped here? ie. "across columns that make up the same rows"

Comment (mfussenegger) [reply blackbox/docs/sql/statements/values.rst]: Hm, I'm not sure, what's meant is that in:

```
   | c1 | c2
r1 | 1  | a
r2 | 2  | b
```

The expressions in c1 must all have the same type, and the same for all expressions within c2. Wasn't sure how to express that better.

Comment (andreidan) [reply blackbox/docs/sql/statements/values.rst]: Hm, I see, it's still very non-descriptive imo. Would  "The expressions executed against each column must have the same type" ?
